device/tt_silicon_driver.cpp

```diff
@@ -313,9 +311,23 @@ namespace {
     };
 }
 // Get TLB index (from zero), check if it's in 16MB, 2MB or 1MB TLB range, and dynamically program it.
-dynamic_tlb set_dynamic_tlb(PCIDevice *dev, unsigned int tlb_index, tt_xy_pair start, tt_xy_pair end,
-                            std::uint64_t address, bool multicast, std::unordered_map<chip_id_t, std::unordered_map<tt_xy_pair, tt_xy_pair>>& harvested_coord_translation, std::uint64_t ordering) {
+dynamic_tlb set_dynamic_tlb(
+    PCIDevice* dev,
+    unsigned int tlb_index,
+    tt_xy_pair start,
+    tt_xy_pair end,
+    std::uint64_t address,
+    bool multicast,
+    std::unordered_map<chip_id_t, std::unordered_map<tt_xy_pair, tt_xy_pair>>& harvested_coord_translation,
+    std::uint64_t ordering)
+{
     auto architecture_implementation = dev->get_architecture_implementation();
+
+    // TODO(Joel): the PCIDevice should not really be carring this around - this
+    // is the last place that extracts it from the PCIDevice.  Since KMD will
+    // eventually take over the TLB programming, this can get removed later on.
+    auto logical_id = dev->get_logical_id();
```

Comment:
Yes, I actually started removing logical_id, but found some other refactors (like moving set_dynamic_tlb) to have more sense to be moved first.
logical_id should be deleted from PCIDevice. I planed to do it, if you don't overtake me in doing it :)

Reply:
That comment is actually a lie - I wrote it before I encountered another instance (`tt_SiliconDevice::broadcast_pcie_tensix_risc_reset` needs the logical id also).  Your observation about other refactors is consistent with mine - too big a blast radius to delete logical_id from PCIDevice in this PR.  I fixed the comment.